extra/modules/greenbids-real-time-data/src/main/java/org/prebid/server/hooks/modules/greenbids/real/time/data/model/data/GreenbidsInferenceDataService.java

```diff
@@ -0,0 +1,183 @@
+package org.prebid.server.hooks.modules.greenbids.real.time.data.model.data;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import com.iab.openrtb.request.BidRequest;
+import com.iab.openrtb.request.Device;
+import com.iab.openrtb.request.Imp;
+import com.maxmind.geoip2.DatabaseReader;
+import com.maxmind.geoip2.exception.GeoIp2Exception;
+import com.maxmind.geoip2.model.CountryResponse;
+import com.maxmind.geoip2.record.Country;
+import org.apache.commons.lang3.StringUtils;
+import org.prebid.server.exception.PreBidException;
+import org.prebid.server.proto.openrtb.ext.request.ExtImpPrebid;
+
+import java.io.IOException;
+import java.net.InetAddress;
+import java.time.ZoneId;
+import java.time.ZonedDateTime;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+public class GreenbidsInferenceDataService {
```

Comment:
I don't like classes placement

the core package is used for the all the core module logic, everything that hooks call
the config package is used for configuration and properties classes

so having all the core services in the model package seems weird for me

please rearrange (don't forget about tests)

Reply:
Have rearranged classes placement

```
# configs and properties
config/
	GreenbidsRealTimeDataConfiguration
	GreenbidsRealTimeDataModule
	GreenbidsRealTimeDataProperties

# core of module with caching logic
core/
	ModelCache
	OnnxModelRunner
	OnnxModelRunnerFactory
	OnnxModelRunnerWithThresholds
	ThresholdCache
	ThrottlingThresholdsFactory

# services and classes used to process each call
model/
	# input data: partner, user agent, inference data from bid request
	data/
		GreenbidsInferenceDataService
		GreenbidsUserAgent
		Partner
		ThrottlingMessage

	# filter service + thresholds/ TPR lists used for filtering proba
	filter/
		FilterService
		ThrottlingThresholds

	# service for crafting result from prediction
	result/
		AnalyticsResult
		GreenbidsInvocationResult
		GreenbidsInvocationService


/v1
	# Impls for analytics tag. Similar as in PbRichMedia
	/model
		ActivityImpl
		AppliedToImpl
		ResultImpl
		TagsImpl
	InvocationResultImplt

	# hook
	GreenbidsRealTimeDataProcessedAuctionRequestHook
```